pkg/collector/worker/worker.go

```diff
@@ -167,8 +167,13 @@ func (w *Worker) Run() {
 			serviceCheckStatus = servicecheck.ServiceCheckCritical
 		}
 
-		if sender != nil && !longRunning && config.Datadog.GetBool("integration_check_status_enabled") {
-			sender.ServiceCheck(serviceCheckStatusKey, serviceCheckStatus, hname, serviceCheckTags, "")
+		if sender != nil {
+			if !longRunning && config.Datadog.GetBool("integration_check_status_enabled") {
```

Comment:
@remeh Should we keep `!longRunning` condition for `sender.Commit()` ? I'm not sure what's the impact of running `sender.Commit()` here for longRunning checks 🤔

Reply:
Fair point... I think it's OK, but let's stick to the exact previous behaviour as it's the goal of this PR. Pushing this immediately